### ISSUE TITLE
Named export 'DefaultApolloClient' not found!

### ISSUE THREAD

**Issue by shamscorner** [antfu/vite-ssg#241]
The following error is happening when building the project (`pnpm build`). However, `pnpm dev` is working fine and graphql is also fetching data properly. You can try with this [Repository](https://github.com/shamscorner/vitesse-stackter-clean-architect) if you want. Everything is set up, just clone and run `pnpm build`.
### App.vue setup
```ts
<script setup lang="ts">
import { DefaultApolloClient } from '@vue/apollo-composable' // -> not found here but it exists
import { apolloClient } from './common/ApolloClient'

// https://v4.apollo.vuejs.org/
provide(DefaultApolloClient, apolloClient)

// https://github.com/vueuse/head
// you can use this to manipulate the document head in any components,
// they will be rendered correctly in the html results with vite-ssg
useHead({
  title: 'Vitesse',
  meta: [
    { name: 'description', content: 'Opinionated Vite Starter Template' },
  ],
})
</script>

<template>
  <RouterView />
</template>
```

### Console error when running `pnpm build`
```
file:///Users/shamscorner/Projects/vitesse-stackter-clean-architect/.vite-ssg-temp/main.mjs:27
import { useQuery, DefaultApolloClient } from "@vue/apollo-composable";
                   ^^^^^^^^^^^^^^^^^^^
SyntaxError: Named export 'DefaultApolloClient' not found. The requested module '@vue/apollo-composable' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from '@vue/apollo-composable';
const { useQuery, DefaultApolloClient } = pkg;

    at ModuleJob._instantiate (node:internal/modules/esm/module_job:127:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:193:5)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:341:24)
    at async importModuleDynamicallyWrapper (node:internal/vm/module:437:15)
    at async Object.build (/Users/shamscorner/Projects/vitesse-stackter-clean-architect/node_modules/.pnpm/vite-ssg@0.19.2_uhne5wm5qpolinapsgjqcugibi/node_modules/vite-ssg/dist/chunks/build.cjs:171:87)
    at async Object.handler (/Users/shamscorner/Projects/vitesse-stackter-clean-architect/node_modules/.pnpm/vite-ssg@0.19.2_uhne5wm5qpolinapsgjqcugibi/node_modules/vite-ssg/dist/node/cli.cjs:25:3)
```

**Comment by userquin** [comment]
@shamscorner can you try adding `format: 'cjs'`  to `ssgOptions`?

**Comment by userquin** [comment]
using `cjs` format won't work, we need to patch some entries on `@apollo/client` packages and change some imports on your repo, stay tuned

**Comment by userquin** [comment]
My fork is using `esno` as dev dependency to allow run `scripts/patch.ts` on node, just install that dev dependency, once you have installed `esno` just run `pnpm install` and you will be able to build your repo.

**Comment by shamscorner** [comment]
Excellent! I really like this solution 😍. It is working as expected.
However, to resolve `@apollo/client`, do we have to add `react` as a dependency on the project?

**Comment by userquin** [comment]
Try removing the dependency, but it seems it is using some react stuff...

**Comment by websitevirtuoso** [comment]
I tried to use this patch but it didn't work for me
I pushed project to review.
https://github.com/websitevirtuoso/vite-ssg-apollo
I am getting error like this
__DEV__ is not defined
![image](https://user-images.githubusercontent.com/5112918/194795449-d45c4c5e-1858-4d2e-95b0-1a28a638a4a0.png)

I pulled your repo and it really works. but my project is much bigger. Hope maybe somebody may assist in it

**Comment by unscatty** [comment]
For anyone still facing this issue I found a workaround, using [pnpm-patch-i](https://github.com/antfu/pnpm-patch-i).

- Run `npx pnpm-patch-i @vue/apollo-composable@<specific version>`, I used version `4.0.0-beta.4`. It's gonna open the editor in the patch folder.
- Edit the file `package.json` located in the root folder of the patch
-  Add 
```json
"type": "module",
"sideEffects": false,
```
to the root of the JSON document, so it ends up something like this:
```json
{
  ...
  "type": "module",
  "sideEffects": false,
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "import": "./dist/index.esm.js",
      "require": "./dist/index.js"
    },
    "./*": "./*"
  },
  ...
}
```

**Comment by unscatty** [comment]
For anyone still facing this issue I found a workaround, after hours struggling, using [pnpm-patch-i](https://github.com/antfu/pnpm-patch-i).

- Run `npx pnpm-patch-i @vue/apollo-composable@<specific version>`, I used version `4.0.0-beta.4`. It's gonna open the editor in the patch folder.
- Edit the `package.json` file located in the root folder of the patch
-  Add 
```json
"type": "module",
"sideEffects": false,
```
to the root of the JSON document, so it ends up something like this:
```js
{
  // ...
  "type": "module",
  "sideEffects": false,
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "import": "./dist/index.esm.js",
      "require": "./dist/index.js"
    },
    "./*": "./*"
  },
  // ...
}
```

- Save the file and go back to the first terminal. Select `yes` to commit the patch
- If you had used the script mentioned above https://github.com/antfu/vite-ssg/issues/241#issuecomment-1118599731
you will have to delete it because it changes exports for apollo libraries. Force install dependencies `pnpm i --force`

Now if you run `pnpm run build` using vite-ssg, it should work.

Tested using:
- Vitesse
- @vue/apollo-composable: `4.0.0-beta.4`
- vite-ssg: `^0.22.1`

Works with pnpm, not sure if it works with other package managers, but there is also https://github.com/ds300/patch-package and process should be the same
